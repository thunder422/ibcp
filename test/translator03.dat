############################################################
#  Translator Test #3: Data Type Assignments (2010-05-09)  #
############################################################

A% = B% + 5
A% = B + 5
A% = B + 5.0
A = B = C = 1.0
A, B, C = 1.0
A, B, C = 1
A%,B% = C% + D%
A% = B% = C% + D%
A$,B$,C$ = "Test"
A$ = B$ = C$ = "Test"

# begin of error tests
Z, Y = A$ + B%
Z, Y = A$ + B$
A,B%,C = 1.0
A,B,C% = 1.0
A%,B,C = 1.0
A,"T",B = 1
A = "T" = B = 1
A,B,C = STR$(1)
A,B,C$ = STR$(1)
A,B,C$ = 1
A,B$,C = 1
A,B,1 = 1
A,B%,1 = 1
1,A,B% = 1
A%,B,C% = 1
A%,B$,C% = 1
A$,B$,C% = 1
A$,B$,C$ = 1
Z = A$ + B * C
A = TAB(4)
A = B + SPC(4)
