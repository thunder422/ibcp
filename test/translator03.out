
Input: A% = B% + 5
Output: TEST:A%<ref> B% 5% +% Assign% 

Input: A% = B + 5
Output: TEST:A%<ref> B 5 + CvtInt Assign% 

Input: A% = B + 5.0
Output: TEST:A%<ref> B 5.0 + CvtInt Assign% 

Input: A = B = C = 1.0
Output: TEST:A<ref> B C = 1.0% =% CvtDbl Assign 

Input: A, B, C = 1.0
Output: TEST:A<ref> B<ref> C<ref> 1.0 AssignList 

Input: A, B, C = 1
Output: TEST:A<ref> B<ref> C<ref> 1 AssignList 

Input: A%,B% = C% + D%
Output: TEST:A%<ref> B%<ref> C% D% +% AssignList% 

Input: A% = B% = C% + D%
Output: TEST:A%<ref> B% C% D% +% =% Assign% 

Input: A$,B$,C$ = "Test"
Output: TEST:A$<ref> B$<ref> C$<ref> "Test" AssignList$ 

Input: A$ = B$ = C$ = "Test"
                      ^^^^^^-- expected numeric expression

Input: Z, Y = A$ + B%
                   ^^-- expected string expression

Input: Z, Y = A$ + B$
              ^^^^^^^-- expected numeric expression

Input: A,B%,C = 1.0
         ^^-- expected double variable

Input: A,B,C% = 1.0
           ^^-- expected double variable

Input: A%,B,C = 1.0
          ^-- expected integer variable

Input: A,"T",B = 1
         ^^^-- expected double variable

Input: A = "T" = B = 1
                 ^-- expected string expression

Input: A,B,C = STR$(1)
               ^^^^^^^-- expected numeric expression

Input: A,B,C$ = STR$(1)
           ^^-- expected double variable

Input: A,B,C$ = 1
           ^^-- expected double variable

Input: A,B$,C = 1
         ^^-- expected double variable

Input: A,B,1 = 1
           ^-- expected double variable

Input: A,B%,1 = 1
         ^^-- expected double variable

Input: 1,A,B% = 1
       ^-- expected item for assignment

Input: A%,B,C% = 1
          ^-- expected integer variable

Input: A%,B$,C% = 1
          ^^-- expected integer variable

Input: A$,B$,C% = 1
             ^^-- expected string item for assignment

Input: A$,B$,C$ = 1
                  ^-- expected string expression

Input: Z = A$ + B * C
                ^^^^^-- expected string expression

Input: A = TAB(4)
           ^^^^-- expected numeric expression

Input: A = B + SPC(4)
               ^^^^-- expected numeric expression

